statsmodels/regression/linear_model.py

```diff
@@ -2912,16 +2921,38 @@ def summary2(
                       xname=xname, yname=yname, title=title)
         smry.add_dict(diagnostic)
 
+        etext = []
+
+        if not self.k_constant:
+            etext.append(
+                "R² is computed without centering (uncentered) since the \
+                model does not contain a constant."
+            )
+        if hasattr(self, 'cov_type'):
+            etext.append(self.cov_kwds['description'])
+        if self.model.exog.shape[0] < self.model.exog.shape[1]:
+            wstr = "The input rank is higher than the number of observations."
+            etext.append(wstr)
+
         # Warnings
         if eigvals[-1] < 1e-10:
             warn = "The smallest eigenvalue is %6.3g. This might indicate that\
             there are strong multicollinearity problems or that the design\
             matrix is singular." % eigvals[-1]
-            smry.add_text(warn)
+            etext.append(warn)
         if condno > 1000:
-            warn = "* The condition number is large (%.g). This might indicate \
-            strong multicollinearity or other numerical problems." % condno
-            smry.add_text(warn)
+            warn = "* The condition number is large (%.g). This might \
+                indicate strong multicollinearity or other numerical \
+                problems." % condno
+            etext.append(warn)
```

Comment:
you could add an `add_text` method to Summary 2 class, so we don't have to know here whether it's string or list concatenation

Reply:
correction: there is already an add_text method

Reply:
The issue here is whether we are sure that every annotation will always be in the form
```
Notes:
[1] Note1
[2] Note2
...
```